src/bwapi/bwresources.py

```diff
@@ -461,37 +461,63 @@ def _name_to_id(self, attribute, setting):
             return setting
 
     def _fill_data(self, data):
-        filled = {}
 
-        if ("name" not in data) or ("includedTerms" not in data):
-            raise KeyError("Need name and includedTerms to post query", data)
-        if self.check_resource_exists(
-            data["name"]
-        ):  # if resource exists, create value for filled['id']
+        default_content_sources = [
+            "qq",
+            "news",
+            "youtube",
+            "forum",
+            "twitter",
+            "review",
+            "facebook",
+            "reddit",
+            "tumblr",
+            "instagram",
+            "blog",
+        ]
+
+        filled = dict()
+
+        if ("name" not in data) or ("booleanQuery" not in data):
+            raise KeyError("Need name and booleanQuery to post query", data)
+
+        filled["booleanQuery"] = data["booleanQuery"]
+
+        # if resource exists, create value for filled['id']
+        if self.check_resource_exists(data["name"]):
             filled["id"] = self.get_resource_id(data["name"])
         if "new_name" in data:
             filled["name"] = data["new_name"]
         else:  # if resource doesn't exist, add name to filled dictionary
             filled["name"] = data["name"]
 
-        filled["includedTerms"] = data["includedTerms"]
-        filled["languages"] = data["languages"] if "languages" in data else ["en"]
-        filled["type"] = data["type"] if "type" in data else "search string"
-        filled["industry"] = (
-            data["industry"] if "industry" in data else "general-(recommended)"
-        )
-        filled["samplePercent"] = (
-            data["samplePercent"] if "samplePercent" in data else 100
-        )
-        filled["languageAgnostic"] = (
-            data["languageAgnostic"] if "languageAgnostic" in data else False
-        )
+        # params with default values
+        filled["type"] = data.get("query_type", "monitor")
+        filled["contentSources"] = data.get("contentSources", default_content_sources)
```

Comment:
I had guessed that this would only work if the `contentSources` kwarg is a list, but it turns out that it can be a single item.  Cool.